ios/RNWebrtcVad.m

```diff
@@ -33,53 +34,86 @@ @implementation RNWebrtcVad {
     self.audioData = nil;
 }
 
-- (dispatch_queue_t)methodQueue
+- (dispatch_queue_t) methodQueue
 {
     return dispatch_get_main_queue();
 }
 
--(instancetype)init {
+-(instancetype) init {
     if (self = [super init]) {
         [AudioInputController sharedInstance].delegate = self;
-        self.audioData = [[NSMutableData alloc] init];
+        cumulativeProcessedSampleLength = 0;
     }
     
     return self;
 }
 
-- (void)dealloc {
+- (void) dealloc {
     voiceDetector = nil;
     self.audioData = nil;
 }
 
 - (void) processSampleData:(NSData *)data
 {
+    if (self.audioData == nil){
+        self.audioData = [[NSMutableData alloc] init];
+    }
+    
     [self.audioData appendData:data];
     
     double sampleRate = [AudioInputController sharedInstance].audioSampleRate;
     
     // Google recommends sending samples (in 10ms, 20, or 30ms) chunk.
     // See: https://github.com/TeamGuilded/react-native-webrtc-vad/blob/master/webrtc/common_audio/vad/include/webrtc_vad.h#L75
     
-    int chunk_size = 0.02 /* seconds/chunk */ * sampleRate * 2 /* bytes/sample */ ; /* bytes/chunk */
+    const double sampleLength = 0.02;
+    
+    cumulativeProcessedSampleLength += sampleLength;
+    int chunkSize = sampleLength /* seconds/chunk */ * sampleRate * 2 /* bytes/sample */ ; /* bytes/chunk */
     
-    if ([self.audioData length] >= chunk_size) {
+    if ([self.audioData length] >= chunkSize) {
 #ifdef DEBUG
         NSLog(@"SENDING = %u", [self.audioData length]);
 #endif
         
         const int16_t* audioSample = (const int16_t*) [self.audioData bytes];
         
-        int isVoice = [voiceDetector isVoice:audioSample sample_rate:sampleRate length:chunk_size/2];
+        int isVoice = [voiceDetector isVoice:audioSample sample_rate:sampleRate length:chunkSize/2];
         
 #ifdef DEBUG
         NSLog(@"Detected Voice %d", isVoice);
 #endif
         
-        // Clear buffer
+        // Clear audio buffer
         [self.audioData setLength:0];
+        
+        // Sends updates ~140ms apart back to listeners
+        const double eventInterval = 0.140;
+        if (cumulativeProcessedSampleLength >= eventInterval) {
+            cumulativeProcessedSampleLength = 0;
+            
+            if (hasListeners) {
+                [self sendEventWithName:@"RNWebrtcVad_SpeakingUpdate" body:@{ @"isVoice": @(isVoice) }];
+            }
+        }
     }
 }
 
-@end
+- (NSArray<NSString *> *) supportedEvents
```

Comment:
this syntax is still kinda throwing me, but this is just saying a pointer to an array of string pointers?

Reply:
Yup, exactly.

Reply:
test